### ISSUE TITLE
LPC845-BRK: Use hardware SPI and TMS RLE mode as JTAG output

### ISSUE THREAD

**Issue by eccoecco** [eccoecco/recyclejtag#5]
Use #4 so that we can bit bash TMS, while TDI and TCK are controlled over hardware SPI to achieve high speeds (probably faster than the serial port can read in, to be honest).

**Comment by eccoecco** [comment]
Technically, since the LPC845 has dual SPI ports, I could actually just put one into master mode, use SCLK (as TCK) and MOSI (as TDI), feed SCLK into the second SPI port, and have its MISO (as TMS), and have TDO fed into MISO of the first SPI port.  It would be hilarious.

**Comment by eccoecco** [comment]
So manually toggled a GPIO pin to act as TMS, and did a simple RLE detection to use LPC845's SPI hardware to output TCK/TDI and clock in TDO.

It worked - but then I found that it would seem that transferring the data back via the virtual serial port sometimes drops data.  Slowing down the data rate (either by manually adding a 10-20ms pause between successive bytes while keeping SPI at max speed, or slowing the SPI port down dramatically) makes the com port send data back.

Now, is this because Win32 has a small receive buffer, and we need to increase the receive buffer size, or the CMSIS-DAP debug probe drops it?

I don't know.  But I think that I won't chase this much further - going to try develop on the STM32 board that I've got.  If this also happens on that board, then we'll look into increasing buffer sizes (as part of #2).
